src/typing/env_js.ml

```diff
@@ -638,12 +748,12 @@ let merge_env =
     match env0, env1, env2 with
     | [], [], [] ->
       ()
-    | scope0 :: env0, scope1 :: env1, scope2 :: env2 -> Scope.(
+    | scope0 :: env0_, scope1 :: env1_, scope2 :: env2_ -> Scope.(
```

Comment:
If you squint, there are a couple cases where we need to revisit the whole list, which is why I changed the names—it's not just to avoid shadowing. In Haskell, it's fairly common to do `x:xs'` in these cases, but a better name/more intention-revealing structure is called for here.